components/fxa-client/src/oauth.rs

```diff
@@ -82,6 +82,54 @@ impl FirefoxAccount {
         Ok(token_info)
     }
 
+    /// Check whether access token is valid using the saved refresh token.
+    /// If there is no refresh token held or if it is not authorized,
+    /// Error is thrown.
+    pub fn check_authorization_status(&mut self, scope: &str) -> Result<IntrospectInfo> {
```

Comment:
API design comment: I don't think we need to pass in a `scope` string here; the response will tell us *all* of the available scopes and the caller can do whatever it likes with that information.